### PR TITLE
feat: add socket and connection timeout configs in redis sink

### PR DESCRIPTION
**Pull request by sumitaich1998**

### CODE REVIEW

File: src/main/java/com/gotocompany/depot/redis/RedisSink.java [hunk 1]
```diff
@@ -47,32 +47,13 @@ public SinkResponse pushToSink(List<Message> messages) {
...
-
-            }
-            retry--;
-        }
-        if (responses == null) {
-            return RedisSinkUtils.getNonRetryableErrors(validRecords, exception, instrumentation);
-        } else {
-            return RedisSinkUtils.getErrorsFromResponse(validRecords, responses, instrumentation);
+        try {
+            responses = redisClient.send(validRecords);
```
Comment (lavkesh) [inline src/main/java/com/gotocompany/depot/redis/RedisSink.java:51]: We should keep the retries.

Comment (sumitaich1998) [reply src/main/java/com/gotocompany/depot/redis/RedisSink.java]: fixed

File: src/main/java/com/gotocompany/depot/redis/client/RedisStandaloneClient.java [hunk 2]
```diff
@@ -31,16 +31,20 @@ public class RedisStandaloneClient implements RedisClient {
     private final DefaultJedisClientConfig defaultJedisClientConfig;
     private final HostAndPort hostAndPort;
     private Jedis jedis;
+    private final int connectionMaxRetries;
+    private final long connectionRetryBackoffMs;
 
-    public RedisStandaloneClient(Instrumentation instrumentation, RedisTtl redisTTL, DefaultJedisClientConfig defaultJedisClientConfig, HostAndPort hostAndPort) {
+    public RedisStandaloneClient(Instrumentation instrumentation, RedisTtl redisTTL, DefaultJedisClientConfig defaultJedisClientConfig, HostAndPort hostAndPort, int connectionMaxRetries, long connectionRetryBackoffMs) {
```
Comment (lavkesh) [inline src/main/java/com/gotocompany/depot/redis/client/RedisStandaloneClient.java:37]: Let's take sinkconfig here, instead of taking so many arguments.

Comment (sumitaich1998) [reply src/main/java/com/gotocompany/depot/redis/client/RedisStandaloneClient.java]: fixed

**Review by lavkesh — APPROVED** [review]

**Pull request merged**